sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -523,4 +563,49 @@ private Span getSpanOrNull(Context azContext) {
 
         return traceContext == null ? null : Span.fromContext(traceContext);
     }
+
+    private SpanKind processKindToSpanKind(ProcessKind processKind) {
+        switch (processKind) {
+            case SEND:
+                return SHARED_SPAN_BUILDER_KIND;
+            case MESSAGE:
+                return SpanKind.PRODUCER;
+            case PROCESS:
+                return SpanKind.CONSUMER;
+            default:
+                return SpanKind.INTERNAL;
+        }
+    }
+
+    private Context startSuppressedSpan(Context context) {
+        return context.addData(SUPPRESSED_SPAN_FLAG, true);
+    }
+
+    private static boolean shouldSuppress(SpanKind kind, Context context) {
+        if (isClientCall(kind)) {
+            boolean suppress = getBoolean(CLIENT_METHOD_CALL_FLAG, context);
+            boolean suppressed = getBoolean(SUPPRESSED_SPAN_FLAG, context);
+
+            return suppress && !suppressed;
```

Comment:
it's not clear to me why checking `!suppressed` ?

in the scenario: `CLIENT` -> `CLIENT` -> `CLIENT`, would this lead to the second `CLIENT` span getting suppressed, but then the third `CLIENT` not getting suppressed?